qmulticast/utils/multipartite_network.py

```diff
@@ -0,0 +1,207 @@
+"""Defines a function to create and return a bipartite source network from a Graph.
+
+TODO maybe make this a subclass of the Network class and build things in init.
+    That'd be neater but not high prority.
+"""
+
+import logging
+from typing import Hashable, Dict, Any, Tuple
+
+import netsquid.qubits.ketstates as ks
+from netsquid.components import QuantumChannel, QuantumProcessor
+from netsquid.components.models.delaymodels import (FibreDelayModel,
+                                                    FixedDelayModel)
+from netsquid.components.models.qerrormodels import (DepolarNoiseModel,
+                                                     FibreLossModel)
+from netsquid.components.qsource import QSource, SourceStatus
+from netsquid.nodes import Network, Node
+from netsquid.qubits.state_sampler import StateSampler
+
+from networkx import DiGraph
+from .functions import gen_GHZ_ket
+
+logger = logging.getLogger(__name__)
+
+
+def unpack_edge_values(node: str, graph: DiGraph) -> Tuple[Hashable, Hashable, Dict[Hashable, Any]]:
```

Comment:
If this is the same as the other file then don't copy it, refer to it instead.

Reply:
done